core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -472,13 +472,21 @@ public void updateTask(TaskResult taskResult) throws Exception {
 
         executionDAO.updateTask(task);
 
+
         //If the task has failed update the failed task reference name in the workflow.
         //This gives the ability to look at workflow and see what tasks have failed at a high level.
         if (FAILED.equals(task.getStatus()) || FAILED_WITH_TERMINAL_ERROR.equals(task.getStatus())) {
             workflowInstance.getFailedReferenceTaskNames().add(task.getReferenceTaskName());
             //In case of a FAILED_WITH_TERMINAL_ERROR the workflow will be terminated and the output of the task is never copied
             //ensuring the task output is copied to the workflow here
             if (FAILED_WITH_TERMINAL_ERROR.equals(task.getStatus())) {
+                //Update the task in the workflow instance
+                Task taskByRefName = workflowInstance.getTaskByRefName(task.getReferenceTaskName());
+                taskByRefName.setStatus(task.getStatus());
+                taskByRefName.setOutputData(task.getOutputData());
+                taskByRefName.setReasonForIncompletion(task.getReasonForIncompletion());
+                taskByRefName.setWorkerId(task.getWorkerId());
+                taskByRefName.setCallbackAfterSeconds(task.getCallbackAfterSeconds());
```

Comment:
Task is already updated at line 473, why is it still needed here?

Reply:
The task is saved to the data store on line 473, the ephemeral state of the workflow needs to be updated to calculate the workflow output. This is needed to determine the workflow output.